File: Lab-1/Project-2/project-2.cpp

#define PIN_LED 2
#define PIN_BUTTON_ON 13
#define PIN_BUTTON_OFF 12
void setup()
{
    pinMode(PIN_LED, OUTPUT);
    pinMode(PIN_BUTTON_ON, INPUT);
    pinMode(PIN_BUTTON_OFF, INPUT);
}

void loop()
{
    if (digitalRead(PIN_BUTTON_ON) == LOW)
    {
        delay(20);
        buttonOn(PIN_LED);
    }
    if (digitalRead(PIN_BUTTON_OFF) == LOW)
    {
        delay(20);
        buttonOff(PIN_LED);
    }
}

void buttonOff(int pin)
{
    digitalWrite(pin, LOW);
}
void buttonOn(int pin)
{
    digitalWrite(pin, HIGH);
}

File: Lab-1/Project-2/project-2.py
import time
from machine import Pin

led = Pin(2, Pin.OUT)
buttonOn = Pin(13, Pin.IN, Pin.PULL_UP)
buttonOff = Pin(12, Pin.IN, Pin.PULL_UP)


def reverseGPIOON():
    led.value(1)

def reverseGPIOOFF():
    led.value(0)

while True:
    if not buttonOn.value():
        time.sleep_ms(20)
        reverseGPIOON()
    if not buttonOff.value():
        reverseGPIOOFF()
        time.sleep_ms(20)
